[DOCS-14641] Add guide page for tracking server-side events in Product Analytics (#37625)
DataDog/documentation: content/en/product_analytics/_index.md

### Track server-side events (API)

> After you set up client-side collection, you can use the [Product Analytics API][21] to send custom events from your server, such as completed checkouts or processed payments. For a step-by-step guide, see [Track Server-Side Events][22].

Reviewer: The linked guide doesn't use numbered steps , suggest to adjust.

```suggestion
After you set up client-side collection, you can use the [Product Analytics API][21] to send custom events from your server, such as completed checkouts or processed payments. For more information, see [Track Server-Side Events][22].
```
